openfisca_core/entities.py

```diff
@@ -297,9 +309,30 @@ def members_position(self):
 
         return self._members_position
 
+    @property
+    def members_entity_id(self):
+        return self._members_entity_id
+
+    @members_entity_id.setter
+    def members_entity_id(self, members_entity_id):
+        self._members_entity_id = members_entity_id
+
+    @property
+    def members_role(self):
+        if self._members_role is None:
+            default_role = self.flattened_roles[0]
```

Comment:
Ok, now I understand better how we can make `role`initialization optional: we are just giving the first role to everyone by default.

I'm not sure whether that make sense: we could easily end up with situations with 5 parents in a family, despite having an entity definition that specifies a max amount to 2... 

Is there really a user demand for a way to initialize simulations with group entities, but without roles? If not, I'd rather just have the `members_role` be mandatory and not make any assumption.

Reply:
This was mostly for unit testing purposes, the earlier tests that built partial functionality could work without roles (and still can, if you pass `None`). I've made it mandatory. We are still making use of the lazy initialization to default roles for the `build_default_simulation` case, however.

Reply:
> 👌